stone/target/python_type_mapping.py

```diff
@@ -0,0 +1,88 @@
+from typing import (
+    Any,
+    Callable,
+    Dict,
+)
+
+if False:  # Versioning issues :(
+    from typing import Type
+    DataTypeCls = Type['DataType']
+else:
+    DataTypeCls = Any
+
+from stone.api import ApiNamespace
+from stone.data_type import (
+    is_boolean_type,
+    is_bytes_type,
+    is_float_type,
+    is_integer_type,
+    is_string_type,
+    is_timestamp_type,
+    is_void_type,
+    DataType,
+    List,
+    Nullable,
+    Alias, UserDefined)
+from stone.target.python_helpers import class_name_for_data_type
+
+# Unfortunately these are codependent, so I'll weakly type the Dict in Callback
+Callback = Callable[[ApiNamespace, DataType, Dict[Any, Any]], str]
+OverrideDefaultTypesDict = Dict[DataTypeCls, Callback]
+
+"""
+NOTE: Due to some trickiness with MyPy, we sometimes choose to do a pure `isinstance` check instead
+of the much nicer looking `is_something_type` check.
+
+Example: in this Alias check, if we used `is_alias_type, you'd get the following message:
```

Comment:
Ack, so unfortunate. The asymmetry is jarring, though it does make sense. Can we continue to use the `is_*` versions but add a cast (https://www.python.org/dev/peps/pep-0484/#casts) in the `elif` block?

Maybe in the future we'll switch entirely to using explicit `isinstance` but that'll be a larger refactor.